kormir/src/lib.rs

```diff
@@ -196,6 +195,161 @@ impl<S: Storage> Oracle<S> {
+        let msg = Message::from_digest(hash.to_byte_array());
+        let announcement_signature = self.secp.sign_schnorr_no_aux_rand(&msg, &self.key_pair);
+
+        let ann = OracleAnnouncement {
+            oracle_event,
+            oracle_public_key: self.public_key(),
+            announcement_signature,
+        };
+        ann.validate(&self.secp).map_err(|_| Error::Internal)?;
+
+        let id = self.storage.save_announcement(ann.clone(), indexes).await?;
+
+        Ok((id, ann))
+    }
+
+    pub async fn sign_numeric_event(
+        &self,
+        id: u32,
+        outcome: i64,
+    ) -> Result<OracleAttestation, Error> {
+        let Some(data) = self.storage.get_event(id).await? else {
+            return Err(Error::NotFound);
+        };
+        if !data.signatures.is_empty() {
+            return Err(Error::EventAlreadySigned);
+        }
+        let descriptor = match &data.announcement.oracle_event.event_descriptor {
+            EventDescriptor::DigitDecompositionEvent(desc) => desc,
+            _ => return Err(Error::Internal),
+        };
+        if descriptor.base != 2 {
+            return Err(Error::Internal);
+        }
+        let max_value = (descriptor.base as i64).pow(descriptor.nb_digits as u32) - 1;
+        let min_value = if descriptor.is_signed { -max_value } else { 0 };
+        if outcome < min_value || outcome > max_value {
+            return Err(Error::InvalidOutcome);
+        }
+
+        let digits = format!(
+            "{:0width$b}",
+            outcome,
+            width = descriptor.nb_digits as usize
+        )
+        .chars()
+        .map(|char| char.to_string())
+        .collect::<Vec<_>>();
+
+        let outcomes = if descriptor.is_signed {
+            let mut sign = vec![if outcome < 0 {
+                "-".to_string()
+            } else {
+                "+".to_string()
+            }];
+            sign.extend(digits);
+            sign
+        } else {
+            digits
+        };
+
+        if data.indexes.len() != outcomes.len() {
+            return Err(Error::Internal);
+        }
+
+        let nonce_keys = data.indexes.iter().map(|i| self.get_nonce_key(*i));
+
+        let mut sigs: Vec<(String, Signature)> = vec![];
+
+        let signatures = outcomes
+            .iter()
+            .zip(nonce_keys)
+            .map(|(outcome, nonce_key)| {
+                let hash = sha256::Hash::hash(outcome.as_bytes());
+                let msg = Message::from_digest(hash.to_byte_array());
+                let sig = dlc::secp_utils::schnorrsig_sign_with_nonce(
+                    &self.secp,
+                    &msg,
+                    &self.key_pair,
+                    &nonce_key.secret_bytes(),
+                );
+                // verify our nonce is the same as the one in the announcement
+                debug_assert!(
+                    sig.encode()[..32] == nonce_key.x_only_public_key(&self.secp).0.serialize()
+                );
```

Comment:
can you change the assert to be the same as this one

https://github.com/benthecarman/kormir/pull/11/files